qtensor/tools/lightcone_orbits.py

```diff
@@ -45,28 +48,36 @@ def graph_cert(G):
     cert = pynauty.certificate(g)
     return cert
 
+def get_cert_e_tuples(e,G=None,p=None):
+    """A helper function for multiprocessing
+    """
+    subgraph = relabel_edge_first(get_edge_subgraph(G, e, p), e)
+    cert = graph_cert(subgraph)
+    return e,cert
+
 
 def get_edge_orbits_lightcones(G,p):
     """Takes graph G and number of QAOA steps p
     returns unique subgraphs that QAOA sees
     dict: {orbit_id : [list of edges in orbit]} 
     and maximum number of nodes in a lightcone subgraph
-    if maxnnodes == G.number_of_nodes(), this simply becomes edge orbits
     """
-    maxnnodes = -1
 
     eorbits = defaultdict(list)
     # for each edge construct the light cone subgraph and compute certificate  
-    for e in G.edges():
-        subgraph = relabel_edge_first(get_edge_subgraph(G, e, p), e)
-        cert = graph_cert(subgraph)
+    if G.number_of_edges() > 1000:
+        # accelerate with multiprocessing if computing for a large graph
+        with Pool(psutil.cpu_count()) as pool:
+            certs_e_tuples = pool.map(partial(get_cert_e_tuples, G=G,p=p), G.edges())
```

Comment:
It's in general a good practice to provide control over what the function is doing. Could you please add the option to control the nubmer of processes? Sometimes it's not very helpful to guess the number of processes, since it can recognize hyper-threads as separate cpus (e.g. on jlse there are 56 physical cores but 112 threads and psutil thinks those are cpus). I think the hard-coded threshold number of edges should be also a function parameter with the reasonable default.

Reply:
Added number of processes as parameter. Now throws a warning if problem is too small instead of switching to serial mode automatically.